### ISSUE TITLE
Discuss bundling filebeat and beats-input into one gem - to replace the file-input.

### ISSUE THREAD

**Issue by guyboertje** [elastic/logstash#5983]
Some ideas to explore (PH and I discussed this v. briefly in Prague):

- The filebeat and the beats input would run on the same machine and not on the edge.
- Logstash would control the filebeat lifecycle (similar to prelert integration if and when).
- Logstash will supply the beats config by translating the file-input user config initially but allow for the filebeat config to be specified directly in the user config with some keys in the `logstash.yml` if it made sense.
- The filebeat executable would ship with the gem.
- The docs for such a dual tech input would be filebeat + beats input docs merge plus clean up.
- Support would more easily support just one setup.
- Our local testing would give us more real world coverage.

/cc @ruflin

**Comment by colinsurprenant** [comment]
if both the LS and FB are running on the same hosts, do we want to still use the lumberjack protocol between both processes? I guess we have a few options:
- keep the same lumberjack protocol and find a way to have it pre-configured using the loopback interface so that it becomes a zero-configuration between LS and FB
- same idea as above but use the http+json used for shipping to ES - which may provide less overhead
- establish a pipe between both processes and use the same json format as above for ES
- come up with a generic stdin/stdout + json protocol to talk with any external processes as plugins as originally proposed in #3755

**Comment by guyboertje** [comment]
AFAIR - Jordan spoke about better bi-directional support for the lumberjack protocol so I suppose we would want to leverage that at some point.

**Comment by colinsurprenant** [comment]
right, but going through the whole lumberjack protocol plus the full network stack for 2 processes on the same host will certainly be less efficient than using a pipe or stdin/out + json protocol. maybe the same bi-directional signalling idea can be reused with that too?
